src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -125,6 +154,11 @@ void PiecewisePolynomial<MaxDeriv>::update(
 
   func[MaxDeriv] = std::move(updated_max_deriv);
   deriv_info_at_update_times_.emplace_back(time_of_update, std::move(func));
+  // While it is technically possible for a different thread to call `update`
+  // during this call, there should never be two different threads trying to
+  // update this object at the same time in our algorithm. An update should come
+  // from only one place and nowhere else so this assignment should be ok
```

Comment:
This comment should be on the previous statement.  The statement it is currently attached to would be fine in such a situation.